clash-prelude/src/Clash/Class/AutoReg/Internal.hs

```diff
@@ -160,6 +173,22 @@ instance AutoReg a => AutoReg (Maybe a) where
        False -> Nothing
   {-# INLINE autoReg #-}
 
+  autoRegHs clk rst en initVal input = createMaybe <$> tagR <*> valR
+   where
+     tag = isJust <$> input
+     tagInit = isJust initVal
+     tagR = suffixName @"tag" register clk rst en tagInit tag
+
+     val = fromMaybe (deepErrorX "autoReg'.val") <$> input
+     valInit = fromMaybe (deepErrorX "autoReg'.valInit") initVal
```

Comment:
```suggestion
     val = fromMaybe (deepErrorX "autoRegHs.val") <$> input
     valInit = fromMaybe (deepErrorX "autoRegHs.valInit") initVal
```

Reply:
And could you also remove the `'`s from the error strings in the original `autoReg`, which I accidentally left in there?